Session/Home/HomeVC.swift

```diff
@@ -203,28 +232,94 @@ final class HomeVC : BaseVC, UITableViewDataSource, UITableViewDelegate, NewConv
             if threads.snapshotOfLastUpdate != firstSnapshot - 1 {
-                return reload() // The code below will crash if we try to process multiple commits at once
+                // Check if we inserted a new message request (if so then unhide the message request banner)
+                if
+                    let extensions: [String: Any] = firstChangeSet[YapDatabaseExtensionsKey] as? [String: Any],
+                    let viewExtensions: [String: Any] = extensions[TSThreadDatabaseViewExtensionName] as? [String: Any]
+                {
+                    // Note: We do a 'flatMap' here rather than explicitly grab the desired key because
+                    // the key we need is 'changeset_key_changes' in 'YapDatabaseViewPrivate.h' so could
+                    // change due to an update and silently break this - this approach is a bit safer
+                    let allChanges: [Any] = Array(viewExtensions.values).compactMap { $0 as? [Any] }.flatMap { $0 }
+                    let messageRequestInserts = allChanges
+                        .compactMap { $0 as? YapDatabaseViewRowChange }
+                        .filter { $0.finalGroup == TSMessageRequestGroup && $0.type == .insert }
+                    
+                    if !messageRequestInserts.isEmpty && UserDefaults.standard[.hasHiddenMessageRequests] {
+                        UserDefaults.standard[.hasHiddenMessageRequests] = false
+                    }
+                }
+                
+                return reload()
             }
         }
+        
         var sectionChanges = NSArray()
         var rowChanges = NSArray()
         ext.getSectionChanges(&sectionChanges, rowChanges: &rowChanges, for: notifications, with: threads)
-        guard sectionChanges.count > 0 || rowChanges.count > 0 else { return }
+        
+        // Separate out the changes for new message requests and the inbox (so we can avoid updating for
+        // new messages within an existing message request)
+        let messageRequestInserts = rowChanges
+            .compactMap { $0 as? YapDatabaseViewRowChange }
+            .filter { $0.finalGroup == TSMessageRequestGroup && $0.type == .insert }
+        let inboxRowChanges = rowChanges
+            .filter { ($0 as? YapDatabaseViewRowChange)?.finalGroup != TSMessageRequestGroup }
+        
+        guard sectionChanges.count > 0 || inboxRowChanges.count > 0 || messageRequestInserts.count > 0 else { return }
+        
         tableView.beginUpdates()
-        rowChanges.forEach { rowChange in
+        
+        // If we need to unhide the message request row and then re-insert it
+        if !messageRequestInserts.isEmpty && UserDefaults.standard[.hasHiddenMessageRequests] {
+            UserDefaults.standard[.hasHiddenMessageRequests] = false
+            tableView.insertRows(at: [IndexPath(row: 0, section: 0)], with: .automatic)
+        }
+        
+        // TODO: Crash due to change from Message Requests getting approved?
```

Comment:
Left TODO here, has this issue been fixed?

Reply:
Ah yep this has been fixed (it's the `rowChange.originalGroup == TSMessageRequestGroup && rowChange.finalGroup == TSInboxGroup` logic) - I'll remove this todo

Reply:
I've pushed up a fix for this